### PR TITLE
CB-13057 : Remove cordova platform save command

### PR DESCRIPTION
**Pull request by audreyso**
### What does this PR do?

Remove cordova platform save command

### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by audreyso** [comment]
Can I delete:
1. commented out tests or parts of tests that contain platforms.json or platform_metadata?
2. Can I delete save.js and save.spec.js?

File: src/cordova/platform_metadata.js [hunk 1]
```diff
@@ -1,111 +1,111 @@
...
-    }
-    var data = getJson(platformJsonFile);
-    delete data[platform];
-    fs.writeFileSync(platformJsonFile, JSON.stringify(data, null, 2), 'utf-8');
-}
-
-module.exports.getPlatformVersions = getVersions;
-module.exports.save = save;
-module.exports.remove = remove;
+// /**
```
Comment (stevengill) [inline src/cordova/platform_metadata.js]: Did you mean to comment out this whole file? Might as well just delete it if that is the case

File: src/cordova/platform/save.js [hunk 2]
```diff
@@ -15,33 +15,35 @@
     under the License.
 */
 
-var semver = require('semver');
-var cordova_util = require('../util');
-var ConfigParser = require('cordova-common').ConfigParser;
-var platformMetadata = require('../platform_metadata');
+// var semver = require('semver');
```
Comment (stevengill) [inline src/cordova/platform/save.js]: Lets delete this whole file as it seems like it is all commented out

File: src/cordova/platform_metadata.js (hunk 1 above)
Comment (stevengill) [reply src/cordova/platform_metadata.js]: Looks like these functions were being used by the save command, hence the comment out

File: src/cordova/platform/remove.js [hunk 3]
```diff
@@ -79,7 +79,7 @@ function remove (hooksRunner, projectRoot, targets, opts) {
             // Remove targets from platforms.json.
             targets.forEach(function (target) {
                 events.emit('verbose', 'Removing platform ' + target + ' from platforms.json file...');
-                platformMetadata.remove(projectRoot, target);
+                // platformMetadata.remove(projectRoot, target);
```
Comment (stevengill) [inline src/cordova/platform/remove.js]: delete

File: src/cordova/platform/remove.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ var events = require('cordova-common').events;
 var npmUninstall = require('cordova-fetch').uninstall;
 var cordova_util = require('../util');
 var config = require('../config');
-var platformMetadata = require('../platform_metadata');
+// var platformMetadata = require('../platform_metadata');
```
Comment (stevengill) [inline src/cordova/platform/remove.js]: delete

File: src/cordova/platform/index.js [hunk 5]
```diff
@@ -80,9 +79,8 @@ function platform (command, targets, opts) {
             return module.exports.update(hooksRunner, projectRoot, targets, opts);
         case 'check':
             return module.exports.check(hooksRunner, projectRoot);
-        case 'save':
-            events.emit('warn', 'This command has been deprecated and will be removed in the next major release of cordova.');
-            return module.exports.save(hooksRunner, projectRoot, opts);
+        // case 'save':
```
Comment (stevengill) [inline src/cordova/platform/index.js]: delete

File: src/cordova/platform/index.js [hunk 6]
```diff
@@ -34,7 +33,7 @@ module.exports.update = function update (hooksRunner, projectRoot, targets, opts
 module.exports.remove = require('./remove');
 module.exports.check = require('./check');
 module.exports.list = require('./list');
-module.exports.save = require('./save');
+// module.exports.save = require('./save');
```
Comment (stevengill) [inline src/cordova/platform/index.js]: delete

File: src/cordova/platform/addHelper.js [hunk 7]
```diff
@@ -220,7 +220,7 @@ function addHelper (cmd, hooksRunner, projectRoot, targets, opts) {
                             // actually installed.
                             var versionToSave = saveVersion ? platDetails.version : spec;
                             events.emit('verbose', 'Saving ' + platform + '@' + versionToSave + ' into platforms.json');
-                            platformMetadata.save(projectRoot, platform, versionToSave);
+                            // platformMetadata.save(projectRoot, platform, versionToSave);
```
Comment (stevengill) [inline src/cordova/platform/addHelper.js]: delete

File: src/cordova/platform/addHelper.js [hunk 8]
```diff
@@ -32,7 +32,7 @@ var cordova_util = require('../util');
 var promiseutil = require('../../util/promise-util');
 var config = require('../config');
 var lazy_load = require('../lazy_load');
-var platformMetadata = require('../platform_metadata');
+// var platformMetadata = require('../platform_metadata');
```
Comment (stevengill) [inline src/cordova/platform/addHelper.js]: delete

File: spec/cordova/platform/save.spec.js [hunk 9]
```diff
@@ -15,57 +15,57 @@
     under the License.
 */
 
-var Q = require('q');
-var rewire = require('rewire');
-var platform_save = rewire('../../../src/cordova/platform/save');
-var platform_metadata = require('../../../src/cordova/platform_metadata');
-var fail;
-var semver = require('semver');
+// var Q = require('q');
```
Comment (stevengill) [inline spec/cordova/platform/save.spec.js]: Delete this file

File: integration-tests/pkgJson-restore.spec.js [hunk 10]
```diff
@@ -142,8 +142,8 @@ describe('tests platform/spec restore with --save', function () {
         var cwd = process.cwd();
         var pkgJsonPath = path.join(cwd, 'package.json');
         var pkgJson;
-        var platformsFolderPath = path.join(cwd, 'platforms/platforms.json');
-        var platformsJson;
+        //var platformsFolderPath = path.join(cwd, 'platforms/platforms.json');
```
Comment (stevengill) [inline integration-tests/pkgJson-restore.spec.js]: delete all the stuff that you commented in this file

File: integration-tests/pkgJson.spec.js [hunk 11]
```diff
@@ -621,15 +621,15 @@ describe('During add, if pkg.json has a platform/plugin spec, use that one.', fu
         var iosVersion;
         var cwd = process.cwd();
         var iosDirectory = path.join(cwd, 'platforms/ios/cordova/version');
-        var platformsFolderPath = path.join(cwd, 'platforms/platforms.json');
+        //var platformsFolderPath = path.join(cwd, 'platforms/platforms.json');
```
Comment (stevengill) [inline integration-tests/pkgJson.spec.js]: delete

File: spec/cordova/platform/addHelper.spec.js [hunk 12]
```diff
@@ -24,7 +24,7 @@ var events = require('cordova-common').events;
 var rewire = require('rewire');
 var platform_addHelper = rewire('../../../src/cordova/platform/addHelper');
 var platform_module = require('../../../src/cordova/platform');
-var platform_metadata = require('../../../src/cordova/platform_metadata');
+// var platform_metadata = require('../../../src/cordova/platform_metadata');
```
Comment (stevengill) [inline spec/cordova/platform/addHelper.spec.js]: delete this + other commented out code

File: spec/cordova/platform/index.spec.js [hunk 13]
```diff
@@ -137,16 +137,16 @@ describe('cordova/platform', function () {
...
-                spyOn(platform, 'save').and.returnValue(true);
-                platform('save', ['android'])
-                    .then(function () {
-                        expect(platform.save).toHaveBeenCalled();
-                    }).fail(function (e) {
-                        expect(e).toBeUndefined();
-                        fail('did not expect fail handler to be invoked');
-                    }).done(done);
-            });
+            // it('should direct `save` commands to the `save` method/module', function (done) {
```
Comment (stevengill) [inline spec/cordova/platform/index.spec.js]: delete

File: spec/cordova/platform/remove.spec.js [hunk 14]
```diff
@@ -21,7 +21,7 @@ var Q = require('q');
 var events = require('cordova-common').events;
 var rewire = require('rewire');
 var platform_remove = rewire('../../../src/cordova/platform/remove');
-var platform_metadata = require('../../../src/cordova/platform_metadata');
+// var platform_metadata = require('../../../src/cordova/platform_metadata');
```
Comment (stevengill) [inline spec/cordova/platform/remove.spec.js]: delete this + other commented out code comments

**Pull request closed without merging**